tools/sdk/ld/eagle.app.v6.common.ld

```diff
@@ -157,6 +157,18 @@ SECTIONS
     *.c.o( EXCLUDE_FILE (umm_malloc.c.o) .literal*, \
            EXCLUDE_FILE (umm_malloc.c.o) .text*)
     *.cpp.o(.literal*, .text*)
+    *.pioenvs\\*\\lib*.a:(EXCLUDE_FILE (umm_malloc.o) .literal*, \
```

Comment:
> That seems to be a library-level switch. Can we apply this switch at framework level?

Currently, is not possible. Need to create `library.json` manifest and make configuration. However, I can open issue for that if you need.

> I.e. .c files get compiled into .c.o, .cpp into .cpp.o and so on? 

It's possible if build each file separately, see [docs](http://www.scons.org/doc/HTML/scons-design/ch04s03.html). We don't play with `env.Object()` and use [env.VariantDir()](https://bitbucket.org/scons/scons/wiki/VariantDir()) where the rest work for us SCons does automatically. Is this critical for you?